main/lsp/lsp_helpers.cc

```diff
@@ -278,14 +284,25 @@ string methodDetail(const core::GlobalState &gs, core::SymbolRef method, core::T
     }
 
     string flagString = "";
-    if (!flags.empty()) {
-        flagString = fmt::format("{}.", fmt::join(flags, "."));
-    }
     string paramsString = "";
-    if (!typeAndArgNames.empty()) {
-        paramsString = fmt::format("params({}).", fmt::join(typeAndArgNames, ", "));
+    if (typeAndArgNames.size() > multilineCutoff) {
+        if (!flags.empty()) {
+            flagString = fmt::format("  {}.\n", fmt::join(flags, ".\n  "));
+        }
+        if (!typeAndArgNames.empty()) {
+            paramsString = fmt::format("  params(\n    {}\n  ).\n", fmt::join(typeAndArgNames, ",\n    "));
+        }
+        return fmt::format("{}{} do\n{}{}  {}\nend", accessFlagString, sigCall, flagString, paramsString,
+                           methodReturnType);
+    } else {
+        if (!flags.empty()) {
+            flagString = fmt::format("{}.", fmt::join(flags, "."));
+        }
+        if (!typeAndArgNames.empty()) {
+            paramsString = fmt::format("params({}).", fmt::join(typeAndArgNames, ", "));
+        }
+        return fmt::format("{}{} {{{}{}{}}}", accessFlagString, sigCall, flagString, paramsString, methodReturnType);
```

Comment:
This could be DRYer. Can you use non-string-literals in `fmt`? I could assign the format strings to vars.

Reply:
(It's arguable that the "wetness" here is acceptable and makes the intent clearer.)

Reply:
This seems fine to me. I think that code golfing it down would make it harder to parse.